src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -422,8 +435,10 @@ public void Reverse()
         /// <param name="count">The number of elements in the range to reverse.</param>
         public void Reverse(int index, int count)
         {
-            ContractUtils.Requires(index >= 0, nameof(index));
-            ContractUtils.Requires(count >= 0, nameof(count));
+            if (index < 0)
+                throw new ArgumentOutOfRangeException(nameof(index));
+            if (count < 0)
+                throw new ArgumentOutOfRangeException(nameof(count));
 
             Array.Reverse(_items, index, count);
```

Comment:
This method has bugs described in https://github.com/dotnet/corefx/issues/14059 due to lack of checks. Will address in separate PR.